### PR TITLE
Address sonar major code smells

### LINKED ISSUE #2334: Fix major code smells
### Description

SonarCloud lists a number of "major" code smells that we should fix to have cleaner code.

### Steps to reproduce

View the [list](https://sonarcloud.io/project/issues?id=hedera-mirror-node&resolved=false&severities=MAJOR) of major code smells in SonarCloud

### Additional context

_No response_

### Hedera network

other

### Version

v0.37.1

### Operating system

_No response_

### PR DESCRIPTION
**Pull request by jnels124**
**Description**:

Address major code smells reported by sonar

**Related issue(s)**:

Fixes #2334

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-common/src/main/java/com/hedera/mirror/common/domain/addressbook/AddressBookEntry.java [hunk 1]
```diff
@@ -36,6 +36,9 @@
 import javax.persistence.JoinColumn;
 import javax.persistence.OneToMany;
 import javax.persistence.Transient;
+
+import com.hedera.mirror.common.exception.NonParsableKeyException;
+
```
Comment (edwin-greene) [inline hedera-mirror-common/src/main/java/com/hedera/mirror/common/domain/addressbook/AddressBookEntry.java:41]: Can optimize imports to move this line down around line 53

File: hedera-mirror-common/src/main/java/com/hedera/mirror/common/util/DomainUtils.java [hunk 2]
```diff
@@ -25,6 +25,9 @@
 import com.google.protobuf.ByteOutput;
 import com.google.protobuf.ByteString;
 import com.google.protobuf.UnsafeByteOperations;
+
+import com.hedera.mirror.common.exception.ProtobufException;
+
```
Comment (edwin-greene) [inline hedera-mirror-common/src/main/java/com/hedera/mirror/common/util/DomainUtils.java:30]: Can optimize imports to move this near line 46.

File: hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/downloader/AbstractDownloaderTest.java [hunk 3]
```diff
@@ -285,15 +290,10 @@ private void startS3Proxy() throws Exception {
...
-            }
-
-            Thread.sleep(1);
-        }
-
-        throw new RuntimeException("Timeout starting S3Proxy, state " + s3Proxy.getState());
+        await("S3Proxy").dontCatchUncaughtExceptions()
+                .atMost(Duration.ofMillis(500))
+                .pollDelay(Duration.ofMillis(1))
+                .until(() -> AbstractLifeCycle.STARTED.equals(s3Proxy.getState()));
```
Comment (edwin-greene) [inline hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/downloader/AbstractDownloaderTest.java:296]: Do we still need to throw an exception at the end of the polling?

File: hedera-mirror-monitor/src/main/java/com/hedera/mirror/monitor/exception/ExpressionConversionException.java [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package com.hedera.mirror.monitor.exception;
+
```
Comment (edwin-greene) [inline hedera-mirror-monitor/src/main/java/com/hedera/mirror/monitor/exception/ExpressionConversionException.java:2]: Needs license

File: hedera-mirror-common/src/main/java/com/hedera/mirror/common/exception/NonParsableKeyException.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package com.hedera.mirror.common.exception;
+
```
[the remark this answers is not in the document]
Comment (jnels124) [reply hedera-mirror-common/src/main/java/com/hedera/mirror/common/exception/NonParsableKeyException.java:2]: fixed

File: hedera-mirror-common/src/main/java/com/hedera/mirror/common/util/DomainUtils.java (hunk 2 above)
Comment (jnels124) [reply hedera-mirror-common/src/main/java/com/hedera/mirror/common/util/DomainUtils.java]: fixed

File: hedera-mirror-monitor/src/main/java/com/hedera/mirror/monitor/exception/ExpressionConversionException.java (hunk 4 above)
Comment (jnels124) [reply hedera-mirror-monitor/src/main/java/com/hedera/mirror/monitor/exception/ExpressionConversionException.java:2]: fixed

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/notify/NotifyingEntityListener.java [hunk 6]
```diff
@@ -99,6 +99,7 @@ public void onCleanup(EntityBatchCleanupEvent event) {
         topicMessages.clear();
     }
 
+    @SuppressWarnings("rawtypes")
     private PreparedStatementCallback callback(Collection<TopicMessage> topicMessages) {
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/notify/NotifyingEntityListener.java:103]:
```suggestion
    private PreparedStatementCallback<int[]> callback(Collection<TopicMessage> topicMessages) {
```

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/exception/MissingCredentialsException.java [hunk 7]
```diff
@@ -20,6 +20,7 @@
  * ‍
  */
 
+@SuppressWarnings("java:S110")
```
Comment (xin-hedera) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/exception/MissingCredentialsException.java:23]: this class, `MissingFileException`, and `ParserSQLException` are not used in any files, we can instead remove them. Note `ParserSQLException` is only used in test cases, so can also delete

Comment (jnels124) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/exception/MissingCredentialsException.java:23]: done

File: hedera-mirror-common/src/main/java/com/hedera/mirror/common/domain/balance/AccountBalanceFile.java [hunk 8]
```diff
@@ -74,12 +74,13 @@ public StreamFile<AccountBalance> copy() {
...
     @Override
-    public Long getConsensusEnd() {
+    public Long getConsensusStart() {
         return consensusTimestamp;
     }
 
     @Override
-    public Long getConsensusStart() {
+    @SuppressWarnings("java:S4144")
+    public Long getConsensusEnd() {
```
[the remark this answers is not in the document]
Comment (jnels124) [reply hedera-mirror-common/src/main/java/com/hedera/mirror/common/domain/balance/AccountBalanceFile.java:82]: fixed

File: hedera-mirror-common/src/main/java/com/hedera/mirror/common/domain/entity/EntityIdEndec.java [hunk 9]
```diff
@@ -34,6 +37,7 @@
  * - 65535 <br/> num: 0 - 4294967295 <br/> Placing entity num in the end has the advantage that encoded ids <=
  * 4294967295 will also be human readable.
  */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
```
[the remark this answers is not in the document]
Comment (jnels124) [reply hedera-mirror-common/src/main/java/com/hedera/mirror/common/domain/entity/EntityIdEndec.java]: nice I didn't know that was a thing.... Fixed

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Pull request merged**